slug.go

```diff
@@ -75,20 +74,16 @@ func packWalkFn(root, src, dst string, tarW *tar.Writer, meta *Meta, dereference
 			return nil
 		}
 
-		// Ignore the .terraform directory itself.
-		if info.IsDir() && info.Name() == ".terraform" {
-			return nil
-		}
-
-		// Ignore any files in the .terraform directory.
-		if !info.IsDir() && filepath.Dir(subpath) == ".terraform" {
+		if m := matchIgnoreRule(subpath, ignoreRules); m {
 			return nil
 		}
 
-		// Skip .terraform subdirectories, except for the modules subdirectory.
-		if strings.HasPrefix(subpath, ".terraform"+string(filepath.Separator)) &&
-			!strings.HasPrefix(subpath, filepath.Clean(".terraform/modules")) {
-			return filepath.SkipDir
+		// Catch directories so we don't end up with empty directories,
+		// the files are ignored correctly
+		if info.IsDir() {
+			if m := matchIgnoreRule(subpath+string(os.PathSeparator), ignoreRules); m {
+				return nil
```

Comment:
I think this should still return `filepath.SkipDir`. The end result is the same, but prevents having to match and exclude every single file recursively in excluded directories. That would be a big difference for directories with a very large number of entries, like `.git/`.

Reply:
The problem/issue with using `SkipDir` is that a child of the ignored directory might be included :/

Reply:
Ah, I wasn't thinking about exceptions to allow paths within a skipped directory. Yeah, in that case I guess we have to check all files regardless.